code/game/objects/items/surgery_tray.dm

```diff
@@ -0,0 +1,207 @@
+	// easy way for us to check if there are any lower quality tools within
+	var/list/recorded_tool_speeds = list()
+	// handle drapes separately so they're always on the bottom
+	if (locate(/obj/item/surgical_drapes) in contents)
+		. += "drapes"
+	// compile all the overlays from items inside us
+	for(var/obj/item/surgery_tool in src)
+		// the overlay we will use if we want to display this one
+		var/actual_overlay = surgery_tool.get_surgery_tool_overlay(tray_extended = !is_portable)
+		if (isnull(actual_overlay))
+			continue // nothing to see here
+
+		// if we don't have tool behaviour then just record the overlay
+		if(!length(surgery_tool.get_all_tool_behaviours()))
+			surgery_overlays[surgery_tool] = actual_overlay
+			continue
+
+		// if we have at least one tool behaviour, check if we already recorded a faster one
+		for (var/surgery_tool_type in surgery_tool.get_all_tool_behaviours())
+			var/highest_speed = LAZYACCESS(recorded_tool_speeds, surgery_tool_type) || INFINITY // bigger number = slower
+			if(surgery_tool.toolspeed > highest_speed)
+				continue
+			// the existing tool was worse than us, ditch it
+			surgery_overlays -= surgery_tool_type
+			LAZYSET(recorded_tool_speeds, surgery_tool_type, surgery_tool.toolspeed)
+			surgery_overlays[surgery_tool_type] = actual_overlay
+
+	for(var/surgery_tool in surgery_overlays)
+		. |= surgery_overlays[surgery_tool]
+
+///Spawn the things we contain on initialisation
+/obj/item/surgery_tray/proc/populate_contents()
+	for (var/thing_path in initial_contents)
+		new thing_path(src)
+	update_appearance(UPDATE_OVERLAYS)
+
+///Sets the surgery tray's deployment state. Silent if user is null.
+/obj/item/surgery_tray/proc/set_tray_mode(new_mode, mob/user)
+	is_portable = new_mode
+	density = !is_portable
+	if(user)
+		user.visible_message(span_notice("[user] [is_portable ? "retracts" : "extends"] [src]'s wheels."), span_notice("You [is_portable ? "retract" : "extend"] [src]'s wheels."))
+
+	if(is_portable)
+		interaction_flags_item |= INTERACT_ITEM_ATTACK_HAND_PICKUP
+		pass_flags |= PASSTABLE
+		RemoveElement(/datum/element/noisy_movement)
+	else
+		interaction_flags_item &= ~INTERACT_ITEM_ATTACK_HAND_PICKUP
+		pass_flags &= ~PASSTABLE
+		AddElement(/datum/element/noisy_movement)
+
+	update_appearance()
+
+/obj/item/surgery_tray/equipped(mob/user, slot, initial)
+	. = ..()
+	if(!is_portable)
+		set_tray_mode(TRUE, user)
+
+/obj/item/surgery_tray/attack_self(mob/user, modifiers)
+	. = ..()
+	if(.)
+		return
+	var/turf/open/placement_turf = get_turf(user)
+	if(isgroundlessturf(placement_turf) || isclosedturf(placement_turf))
+		balloon_alert(user, "can't deploy!")
+		return TRUE
+	if(!user.transferItemToLoc(src, placement_turf))
+		balloon_alert(user, "tray stuck!")
+		return TRUE
+	set_tray_mode(FALSE, user)
+	return
+
+/obj/item/surgery_tray/attack_hand(mob/living/user)
+	if(!user.can_perform_action(src, NEED_HANDS))
+		return ..()
+	var/obj/item/grabbies = pick(contents)
```

Comment:
Oh, if you truly can't find a good way to do something like this without overriding attack_hand(), at least allow the player to choose the item through a (persistent) radial menu instead of picking a random tool from the bunch. That'd definitely prove more useful.

Reply:
it's currently a slightly useless feature, but is important groundwork for a feature jacquerel and i are interested in, outside the scope of this PR, where op tables can be synced with the table to have a fluid draw mechanic. getting players used to the radial and then replacing it with fumble again once that comes would probably be confusing, so i'd like to keep it on attack_hand() for now if that's fine